File: frontend/demo/component/details/details-basic.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/details';
import '@vaadin/vertical-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('details-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    // tag::snippet[]
    return html`
      <vaadin-details summary="Contact information" opened>
        <vaadin-vertical-layout>
          <span>Sophia Williams</span>
          <span>sophia.williams@company.com</span>
          <span>(501) 555-9128</span>
        </vaadin-vertical-layout>
      </vaadin-details>
    `;
    // end::snippet[]
  }
}

File: src/main/java/com/vaadin/demo/component/details/DetailsBasic.java
package com.vaadin.demo.component.details;

import com.vaadin.flow.component.details.Details;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Span;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("details-basic")
public class DetailsBasic extends Div {

  public DetailsBasic() {
    // tag::snippet[]
    Span name = new Span("Sophia Williams");
    Span email = new Span("sophia.williams@company.com");
    Span phone = new Span("(501) 555-9128");
    
    VerticalLayout content = new VerticalLayout(name, email, phone);
    content.setSpacing(false);
    content.setPadding(false);

    Details details = new Details("Contact information", content);
    details.setOpened(true);

    add(details);
    // end::snippet[]
  }

  public static class Exporter extends DemoExporter<DetailsBasic> { // hidden-source-line
  } // hidden-source-line
}
